providers/hexonet/hexonetProvider.go

```diff
@@ -0,0 +1,77 @@
+// Package hexonet implements a registrar that uses the hexonet api to set name servers. It will self register it's providers when imported.
+package hexonet
+
+// HEXONET is a leading developer and operator of domain names and DNS platforms.
+// Thousands of individuals, service providers and registrars globally choose HEXONET
+// for domains and DNS for our advanced technology, operational performance and up-time,
+// and most importantly for unparalleled expertise in DNS.
+// DnsControl with HEXONET's DNS marries DNS automation with an industry-leading DNS
+// platform that supports DNSSEC, PremiumDNS via Anycast Network, and nearly all of
+// DnsControl's listed provider features.
+
+import (
+	"encoding/json"
+
+	"github.com/StackExchange/dnscontrol/providers"
+	hxcl "github.com/hexonet/go-sdk/client"
+	"github.com/pkg/errors"
+)
+
+// HXClient describes a connection to the hexonet API.
+type HXClient struct {
+	APILogin    string
+	APIPassword string
+	APIEntity   string
+	client      *hxcl.Client
+}
+
+var features = providers.DocumentationNotes{
+	providers.CanUseAlias:            providers.Can("We fallback to CNAME up to now. Our API comes already with realtime ALIAS support over X-ALIAS-A / X-ALIAS-AAAA. This will be supported shortly."),
+	providers.CanUseCAA:              providers.Can(),
+	providers.CanUsePTR:              providers.Can(),
+	providers.CanUseRoute53Alias:     providers.Cannot(),
+	providers.CanUseSRV:              providers.Can(),
+	providers.CanUseTLSA:             providers.Can(),
+	providers.CanUseTXTMulti:         providers.Can(),
+	providers.CantUseNOPURGE:         providers.Can(),
+	providers.DocCreateDomains:       providers.Can(),
+	providers.DocDualHost:            providers.Can(),
+	providers.DocOfficiallySupported: providers.Can(),
+}
+
+func newProvider(conf map[string]string) (*HXClient, error) {
+	api := &HXClient{
+		client: hxcl.NewClient(),
+	}
+	api.APILogin, api.APIPassword, api.APIEntity = conf["apilogin"], conf["apipassword"], conf["apientity"]
```

Comment:
Perhaps we should default "apientity" to the live system to save anyone needing to know the magic "54cd" identifier?

Perhaps even switch it to a "use OTE" type flag and handle the entity ids internally only?

Reply:
Yes, that how we already cover it in our connector library `go-sdk`. By default live system will be used. Just in case customer wants to use OT&E system, method api.client.UseOTESystem() needs to be called.
The intention of the methods UseOTESystem and UseLIVESystem was exactly to not force people to use this very technical entity values. My integration here was a quickshot with a planned later rewrite - that I've forgotten to apply. Thanks for pointing on that.